src/Compiler/Driver/GraphChecking/DependencyResolution.fs

```diff
@@ -235,20 +236,47 @@ let mkGraph (filePairs: FilePairMap) (files: FileInProject array) : Graph<FileIn
                 | None -> Array.empty
                 | Some sigIdx -> Array.singleton sigIdx
 
+            let wrongOrderSignature =
```

Comment:
Sorry to mention this too late, but would be great to have a test in https://github.com/dotnet/fsharp/blob/main/tests/FSharp.Compiler.ComponentTests/TypeChecks/Graph/DependencyResolutionTests.fs that covers this.

From the code alone, I can only guess what this does.